ingestion/src/metadata/utils/streamable_logger.py

```diff
@@ -0,0 +1,506 @@
+            except Exception as e:
+                logger.warning(f"Error closing previous handler: {e}")
+        cls._instance = handler
+
+    @classmethod
+    def cleanup(cls) -> None:
+        """Clean up the current handler"""
+        if cls._instance:
+            try:
+                metadata_logger = logging.getLogger("metadata")
+                metadata_logger.removeHandler(cls._instance)
+                cls._instance.close()
+                logger.debug("Streamable logging handler cleaned up")
+            except Exception as e:
+                logger.warning(f"Error during handler cleanup: {e}")
+            finally:
+                cls._instance = None
+
+
+def setup_streamable_logging_for_workflow(
+    metadata: OpenMetadata,
+    pipeline_fqn: Optional[str] = None,
+    run_id: Optional[UUID] = None,
+    log_level: int = logging.INFO,
+    enable_streaming: bool = False,
+) -> Optional[StreamableLogHandler]:
+    """
+    Setup streamable logging for a workflow execution.
+    This is automatically called when a workflow starts if:
+    1. The IngestionPipeline has enableStreamableLogs set to true
+    2. The server has log storage configured
+    3. The pipeline FQN and run ID are available
+
+    Args:
+        metadata: OpenMetadata client instance
+        pipeline_fqn: Fully qualified name of the pipeline
+        run_id: Unique run identifier
+        log_level: Logging level
+        enable_streaming: Whether to enable streaming (from IngestionPipeline config)
+
+    Returns:
+        StreamableLogHandler instance if configured, None otherwise
+    """
+    # Check if we have the required parameters
+    if not enable_streaming or not pipeline_fqn or not run_id:
+        logger.debug(
+            f"Streamable logging not configured: enable={enable_streaming}, "
+            f"pipeline_fqn={pipeline_fqn}, run_id={run_id}"
+        )
+        return None
+
+    try:
+        # Check if server supports log storage by trying to get the configuration
+        # This would need to be implemented as an API endpoint
+        # For now, we'll assume it's enabled if the env var is set
+
+        # Clean up any existing handler
+        existing_handler = StreamableLogHandlerManager.get_handler()
+        if existing_handler:
+            existing_handler.close()
+
+        # Create and configure the handler
+        handler = StreamableLogHandler(
+            metadata=metadata,
+            pipeline_fqn=pipeline_fqn,
+            run_id=run_id,
+            enable_streaming=True,
+        )
+
+        # Set formatter
+        formatter = logging.Formatter(
+            "[%(asctime)s] %(levelname)-8s {%(name)s:%(module)s:%(lineno)d} - %(message)s",
+            datefmt="%Y-%m-%d %H:%M:%S",
+        )
+        handler.setFormatter(formatter)
+        handler.setLevel(log_level)
+
+        # Add handler to the metadata logger (parent of all ingestion loggers)
+        metadata_logger = logging.getLogger("metadata")
```

Comment:
```suggestion
        metadata_logger = logging.getLogger(METADATA_LOGGER)
```

you'll need to do the import from logger.py